gcloud-java-bigquery/src/test/java/com/google/gcloud/bigquery/CopyJobInfoTest.java

```diff
@@ -0,0 +1,139 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.google.gcloud.bigquery;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertTrue;
+
+import com.google.common.collect.ImmutableList;
+import com.google.gcloud.bigquery.JobInfo.CreateDisposition;
+import com.google.gcloud.bigquery.JobInfo.WriteDisposition;
+
+import org.junit.Test;
+
+import java.util.List;
+
+public class CopyJobInfoTest {
+
+  private static final TableId SOURCE_TABLE = TableId.of("dataset", "sourceTable");
+  private static final List<TableId> SOURCE_TABLES = ImmutableList.of(
+      TableId.of("dataset", "sourceTable1"),
+      TableId.of("dataset", "sourceTable2")
+  );
+  private static final TableId DESTINATION_TABLE = TableId.of("dataset", "destinationTable");
+  private static final CreateDisposition CREATE_DISPOSITION = CreateDisposition.CREATE_IF_NEEDED;
+  private static final WriteDisposition WRITE_DISPOSITION = WriteDisposition.WRITE_APPEND;
+  private static final JobId JOB_ID = JobId.of("job");
+  private static final JobStatus JOB_STATUS = new JobStatus(JobStatus.State.DONE);
+  private static final JobStatistics JOB_STATISTICS = JobStatistics.builder()
+      .creationTime(1L)
+      .endTime(3L)
+      .startTime(2L)
+      .build();
+  private static final CopyJobInfo COPY_JOB =
+      CopyJobInfo.builder(DESTINATION_TABLE, SOURCE_TABLE)
+          .jobId(JOB_ID)
+          .status(JOB_STATUS)
+          .createDisposition(CREATE_DISPOSITION)
+          .writeDisposition(WRITE_DISPOSITION)
+          .dryRun(true)
+          .statistics(JOB_STATISTICS)
+          .build();
+  private static final CopyJobInfo COPY_JOB_INFO_MULTIPLE_TABLES =
+      CopyJobInfo.builder(DESTINATION_TABLE, SOURCE_TABLES)
+          .jobId(JOB_ID)
+          .status(JOB_STATUS)
+          .createDisposition(CREATE_DISPOSITION)
+          .writeDisposition(WRITE_DISPOSITION)
+          .build();
+
+  @Test
+  public void testToBuilder() {
+    compareCopyJobInfo(COPY_JOB, COPY_JOB.toBuilder().build());
+    compareCopyJobInfo(COPY_JOB_INFO_MULTIPLE_TABLES,
+        COPY_JOB_INFO_MULTIPLE_TABLES.toBuilder().build());
+    CopyJobInfo job = COPY_JOB.toBuilder()
+        .destinationTable(TableId.of("dataset", "newTable"))
+        .build();
+    assertEquals("newTable", job.destinationTable().table());
+    job = job.toBuilder().destinationTable(DESTINATION_TABLE).build();
+    compareCopyJobInfo(COPY_JOB, job);
+  }
+
+  @Test
+  public void testToBuilderIncomplete() {
+    CopyJobInfo job = CopyJobInfo.of(DESTINATION_TABLE, SOURCE_TABLES);
+    compareCopyJobInfo(job, job.toBuilder().build());
+  }
+
+  @Test
+  public void testBuilder() {
+    assertEquals(JOB_ID, COPY_JOB_INFO_MULTIPLE_TABLES.jobId());
```

Comment:
If we do set `JobInfo` fields (mentioned in my comment above), then we could also add assert those fields here.